brave/brave-ios
# Fix #7032: Voice Search

## Summary of Changes

This pull request is implementing Voice Search which will use on device speech recognition will trigger a search query in the browser.

PR is adding 2 entry points for starting the Speech Recognition and displaying the UI. 

First point of entry is when Search bar is activated and emptied (location QR code is displayed)
Second point of entry is on NTP when URL bar is empty.

When Voice search is activated AVAudio Engine is created and Audio Session input node will be added as SFSpeech audio buffer.

Security Ticket: https://github.com/brave/security/issues/1311 - done


This pull request fixes #7032

## Submitter Checklist:


## Test Plan:
- Start the Voice Search from one of the entry points.
- Check idle animation
- Enter search query with voice
- Check search query is transferred to search engine 

Notes for QA: Additionally we have to check - test this feature with background audio and playlist.

## Screenshots:


https://github.com/brave/brave-ios/assets/6643505/47fdf007-9632-4c32-9279-68aa3b99335e

![longsearch1](https://github.com/brave/brave-ios/assets/6643505/f0b16ef8-38e3-43c5-bc3d-1c2ef0cf67a5)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
